### ISSUE TITLE
[enhancement]: Change brush-size like in Photoshop

### ISSUE THREAD

**Issue by BeeMeeN** [invoke-ai/InvokeAI#5579]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Contact Details

info@nieschalk.com

### What should this feature add?

Add the possibility to interactively change brush-sizees in unified canvas with keyboard shortcut.
Hold alt+right mousebutton, then move left-right to change brush-size quickly and easily like in Photoshop. The current size should be previewed directly on the canvas to see if it fits.
I've been missing this feature since the beginning, maybe because I'm so used to use it in Photoshop...

**Comment by ufuksarp** [comment]
> @ufuksarp
> 
> ctrl/meta plus scroll to adjust brush size was just implemented.
> 
> Is hold a key and just move the mouse better than hold a key and scroll?
> 
> I'm not sure which is convention or most useful

I can’t say. But I think it’s better to continue with what’s already implemented. Implementing the opacity adjustment with an extra key like `shift` on top of current keys for brush size should already be miles ahead than bracket keys.

**Comment by BeeMeeN** [comment]
Hi, I tried, but in my Version 3.6.2 it hasn't been implemented yet. But to be honest, I would prefer the alt/cmd + move the mouse, it feels much more familiar because it would be the same combination as is in Photoshop.

**Comment by BeeMeeN** [comment]
@ufuksarp I disagree ;) I think it would be better to offer key and mouse shortcuts that most (pro) users are familiar with from their daily work (with Photoshop)...

**Comment by ufuksarp** [comment]
> @ufuksarp I disagree ;) I think it would be better to offer key and mouse shortcuts that most (pro) users are familiar with from their daily work (with Photoshop)...

Sure, configurable hotkeys are the best option, no doubt there. But I’m not sure if they want to take on that currently. It’s going to be a lot of work I feel. Again, I can’t say what configuration is the best/most conventional one, maybe Discord community can be polled with pre set options? @Millu

**Comment by BeeMeeN** [comment]
@psychedelicious @ufuksarp

Hey guys, hope you are fine. I just installed the latest RC and tried the **new ctrl+mousewheel-function** to resize the brush! Now I know why Adobe decided **not to use** the mousewheel. Because of the **mousewheel-interpolation for e. g. "smooth scrolling"**!!! When I now press ctrl and use the wheel, my brushsize changes without any chance to do a fine adjustment. It has no fine gradation, it either starts **inflating** like a balloon **to the maximum** value or it starts **shrinking** to the **smallest** setting, making it look as if the brush symbol is animated and becomes large or small, but nothing between.
That's why I would still prefer the Photoshop-method with right mousebutton and then just moving to right or left. 🤓
What do you think? And what was your experience?

**Comment by psychedelicious** [comment]
@BeeMeeN Yes, it's a tricky problem to solve. You can see a couple attempts to implement a smooth algorithm in the PR. I did more experiments locally too. 

It's possible to make one that works well for either trackpad (which sends wheel events very frequently) or scroll wheel (which sends one one event per "click" of the wheel). Each device needs a different algorithm. Trackpad can use the magnitude of the wheel events (deltaY) while scroll wheel needs you to track the frequency of the wheel events and use that to determine the change in brush size. 

Browsers do not differentiate between trackpad scroll and scroll wheel in a sane way. They set the ctrlKey event attribute to true for trackpad, even if ctrl is not held.

This makes it difficult to be sure which device is being used and decide which algorithm to use. 

The mouse move approach makes sense to me bc it uses the same event for both devices that has very similar technical characteristics regardless of device.

**Comment by BeeMeeN** [comment]
@psychedelicious

> The mouse move approach makes sense to me bc it uses the same event for both devices that has very similar technical characteristics regardless of device.

Unfortunately, I'm not a programmer, but it seems to me that the solution with the scroll wheel seems to be very complicated. Even though I'm repeating myself again, the alt + rmb and just move left or right solution seems very simple compared to it 😬
